### ISSUE TITLE
Exported gallery uses wrong image when image and video have same name

### ISSUE THREAD

**Issue by tribut** [thumbsup/thumbsup#111]
## Bug report

~~~
$ ls in/
test.jpg  test.mp4
$ thumbsup --input in --output out
~~~

The resulting gallery has two items - an image and a video - as expected, but the image is the generated preview of `test.mp4` instead of the contents of `test.jpg`.

**Comment by rprieto** [comment]
Thanks! I can definitely see why that’s happening.

Unfortunately the only way I can think of (since [target paths](https://github.com/thumbsup/thumbsup/blob/master/src/model/output.js) don’t know about each other) is to rename video poster files, e.g. from `media/large/videoname.jpg` to `media/large/videoname.poster.jpg`. This would mean re-building all posters when you install the latest version.

Any other ideas?

**Comment by tribut** [comment]
Hm, I guess the only way to be sure there would be no conflicts is to use different output paths. `media/large-video/foo.jpg` or something like that?

**Comment by rprieto** [comment]
Quick update: I started implementing a fix and noticed it also happens with the thumbnail. Effectively:

| | IMG_0001.jpg | IMG_0001.mp4 |
|---|---|---|
| thumbnail | thumbs/IMG_0001.jpg | thumbs/IMG_0001.jpg |
| large | large/IMG_0001.jpg | large/IMG_0001.jpg |
| video | - | large/IMG_0001.mp4 |
| download | original/IMG_0001.jpg | original/IMG_0001.mp4 |

I think it will be hard to solve without making a mess at the moment. I think it should be combined with #32 to review output paths in general, for example supporting

```js
// photo
media/IMG_0001.thumb.jpg
media/IMG_0001.large.jpg
// video
media/IMG_0001.videothumb.jpg
media/IMG_0001.poster.jpg
media/IMG_0001.resized.mp4
```

**Comment by tribut** [comment]
Good thinking. Maybe including the original extension would be a good idea to avoid clashes when, for example, `IMG_0001.mp4` and `IMG_0001.mov` exist? Such as `media/IMG_0001.mov_resized.mp4`?

**Comment by tribut** [comment]
If we really want to be sure I guess we would have to use different paths though and use the original filenames. Otherwise someone having source files named `IMG_0001.jpg` and `IMG_0001.jpg_thumb.jpg` would run into a problem. Granted, that is a very constructed example, but just wanted to make sure I mentioned it...

**Comment by rprieto** [comment]
Hi @tribut . I just implemented a new optional output structure based on the idea above. This fixes #32 (file structure is easier to navigate), but it also fixes the problem of filename clashes.

Do you mind having a look at the pull-request (https://github.com/thumbsup/thumbsup/pull/143/files) to see if it matches what you had in mind? The files `src/model/structure.js` and `src/model/structure.spec.js` should be pretty explicit.

**Comment by tribut** [comment]
Yes, that does look what I imagined. Great work!
